.github/workflows/benchmark.yml

```
@@ -0,0 +1,130 @@
+
+            -   name: Use GCC-10 on ubuntu as default
+                if: ${{ runner.os == 'Linux' }}
+                run: |
+                    sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-10 10
+                    sudo update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-10 10
+
+            -   name: Get Conan configuration
+                run: |
+                    conan profile new default --detect
+                    conan config install https://github.com/Ultimaker/conan-config.git
+
+            -   name: Install dependencies
+                run: conan install . ${{ needs.conan-recipe-version.outputs.recipe_id_full }} -o enable_benchmarks=True -s build_type=Release --build=missing --update -g GitHubActionsRunEnv -g GitHubActionsBuildEnv
+
+            -   name: Upload the Dependency package(s)
+                run: conan upload "*" -r cura --all -c
+
+            -   name: Set Environment variables from Conan install (bash)
+                if: ${{ runner.os != 'Windows' }}
+                run: |
+                    . ./activate_github_actions_runenv.sh
+                    . ./activate_github_actions_buildenv.sh
+                working-directory: build/generators
+
+            -   name: Build CuraEngine and tests
+                run: |
+                    cmake --preset release
+                    cmake --build --preset release
+
+            -   name: Run benchmark CuraEngine
+                id: run-test
+                run: ./benchmarks --benchmark_format=json --benchmark_context=version=`${{ needs.conan-recipe-version.outputs.recipe_semver_full }}` | tee benchmark_result.json
+                working-directory: build/Release/benchmark
+
+            -   name: Store benchmark result
+                uses: benchmark-action/github-action-benchmark@v1
+                with:
+                    name: C++ Benchmark
+                    tool: 'googlecpp'
+                    output-file-path: build/Release/benchmark/benchmark_result.json
+                    github-token: ${{ secrets.GITHUB_TOKEN }}
+                    auto-push: true
+                    # Show alert with commit comment on detecting possible performance regression
+                    alert-threshold: '200%'
```

I'm not sure how much the runners performance affects this, but I feel like going for a slightly lower threshold would be good and we can up it if we have false positives. 

```suggestion
                    alert-threshold: '150%'
```

---

According to the doc the runners can vary somewhere around the 20%. That is why I would first get it to run on each push commit to main, to see the actual variance.

I'm fine with a lower value, btw that would allow us to test that  functionality as well